src/main/java/org/opensearch/performanceanalyzer/PerformanceAnalyzerApp.java

```diff
@@ -250,23 +254,71 @@ private static void startReaderThread(
+                                            "Error in ReaderMetricsProcessor...restarting, retryCount left: {}."
+                                                    + "Exception: {}",
+                                            retryAttemptLeft,
+                                            e.getMessage());
                                     StatsCollector.instance()
                                             .logException(
                                                     StatExceptionCode.READER_RESTART_PROCESSING);
+
+                                    if (TroubleshootingConfig.getEnableDevAssert()) break;
+
+                                    // All retry attempts exhausted; handle thread failure
+                                    if (retryAttemptLeft <= 0) handleReaderThreadFailed();
                                 }
                             }
                         },
                         PerformanceAnalyzerThreads.PA_READER);
-
         readerThread.start();
     }
 
+    private static void handleReaderThreadFailed() {
+        // Reader subcomponent is responsible for processing, cleaning metrics written by PA.
+        // Since Reader thread fails to start successfully, execute following:
+        //
+        // 1. Disable PA - Stop collecting OpenSearch metrics
+        // 2. Terminate RCA Process - Gracefully shutdown all
+        //    existing resources/channels, including Reader.
+        try {
+            LOG.info(
+                    "Exhausted {} attempts - unable to start Reader Thread successfully; disable PA",
+                    READER_RESTART_MAX_ATTEMPTS);
+            disablePA();
+            LOG.info("Attempt to disable PA succeeded.");
+            StatsCollector.instance()
+                    .logException(StatExceptionCode.READER_ERROR_PA_DISABLE_SUCCESS);
+        } catch (Throwable e) {
+            LOG.info("Attempt to disable PA failing: {}", e.getMessage());
+            StatsCollector.instance()
+                    .logException(StatExceptionCode.READER_ERROR_PA_DISABLE_FAILED);
+        } finally {
+            cleanupAndExit();
+        }
+    }
+
+    private static void disablePA() {
+        String PA_CONFIG_PATH = Util.PA_BASE_URL + "/cluster/config";
+        String PA_DISABLE_PAYLOAD = "{\"enabled\": false}";
+
+        int resCode =
+                ESLocalhostConnection.makeHttpRequest(
+                        PA_CONFIG_PATH, HttpMethod.POST, PA_DISABLE_PAYLOAD);
+        if (resCode != HttpURLConnection.HTTP_OK) {
+            throw new RuntimeException("Failed to disable PA");
+        }
+    }
+
+    private static void cleanupAndExit() {
+        LOG.info("Reader thread not coming up successfully - Shutting down RCA Runtime");
+        StatsCollector.instance().logException(StatExceptionCode.READER_ERROR_RCA_AGENT_STOPPED);
+
+        // Terminate Java Runtime, executes {@link #shutDownGracefully(ClientServers clientServers)}
+        System.exit(1);
```

Comment:
In case users want(or try) to enable PA/RCA back via CLI dynamically, I guess it wouldn't work?
How do we handle such scenarios?

Does it make sense if we just terminate the reader thread? As kill rca process, disabling PA plugin all seems too intrusive.

Reply:
I disagree, killing only the PA thread while keeping the process idle is wasting the system resource. The RCA process as we know depends on reader sub-compoent for providing the data, which then flows throught analysis graph and made available to nodes/users via the grpc and web server resp. 

> In case users want(or try) to enable PA/RCA back via CLI dynamically, I guess it wouldn't work?
How do we handle such scenarios?

PA can be enabled using the REST API call, while starting RCA will require bringing up the RCA Agent via the [performance-analyzer-agent](https://github.com/opensearch-project/performance-analyzer-rca/blob/main/bin/performance-analyzer-agent) tool. This updated behavior will be added to the documentation. 

Let me know your throughts.